[process-agent] Remove pkg/process/config package

xornivore opened the pull request:
### What does this PR do?

- Removes pkg/process/config package
- Move config bootstrap helpers into cmd/process-agent/command (for now)
- Provides env var bindings for legacy orchestrator settings
- Adds tests to pkg/orchestrator/config for legacy orchestrator settings

### Motivation

Remove pkg/process/config in preparation for using components.

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

Validate functionality related to loading configuration files and env vars in the process agent.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

xornivore committed:
[process-agent] Remove pkg/process/config package

JLineaweaver reviewed (approved):
Looks like a good cleanup to me.

just-chillin reviewed (approved):
Great refactor!

kkhor-datadog commented on cmd/process-agent/command/command.go:
```diff
@@ -106,18 +110,89 @@ func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCm
+	}
+
+	_, err := os.Stat("/host")
+	if !os.IsNotExist(err) {
```
Not sure how much of an impact this has, but it seems like there is a small change in behavior - previously we were using `util.PathExists`, which return true only if there was no error returned from `os.Stat`, whereas here we continue if `!os.IsNotExist`. In particular it seems like if the error was something like a permission error we'd see different behavior compared to the previous implementation which would not set these environment variables, whereas this one would

xornivore replied:
Yep, thank you, nice catch! To think of it, the helper exists in the util which will still be around for some time, no reason not to use it.

kkhor-datadog replied:
Sounds good, looks like there is another occurrence of this in `loadConfigIfExists` below as well

xornivore committed:
Address review feedback from @kkhor-datadog

- Revert back to using util.PathExists for simplicity
- Clean up code with early exits

sgnn7 reviewed (approved):
LGTM with a few nits though it would be good to have @ogaca-dd or @truthbk take a look as well.

xornivore committed:
Review feedback from @sgnn7

xornivore commented on cmd/process-agent/command/command.go:
```diff
@@ -106,18 +111,82 @@ func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCm
+		return err
+	}
+
+	var (
```
I kept them closer to where they were initialized to the config as opposed to the beginning of the func where we have a couple of early exit cases. 

Regarding the config defaults, while the defaults do make sense for the daemon execution, for one shot commands they may not.
